test-result-summary-client/src/Build/GitNewIssue.jsx

```diff
@@ -0,0 +1,120 @@
+import renderDuration from './Duration';
+
+import './table.css';
+
+import moment from 'moment';
+const DAY_FORMAT = 'MMM DD YYYY, hh:mm a';
+
+export default class GitNewissue extends Component {
+    state = {
+        urlParams: null,
+        body: '',
+        title: '',
+    };
+
+    async componentDidMount() {
+        await this.updateData();
+    }
+
+    async updateData() {
+        const { testId, testName, duration, buildId, buildName, 
+            buildUrl, machine, buildTimeStamp, javaVersion, testResult } = getParams( this.props.location.search );
+            
+        let firstSeenFailure, firstFailDate, firstFailUrl, firstFailJavaVersion;
+        let failCount = 0;
+        let machinesMap = {};
+        let machines = new Set();
+        const fetchBuild = await fetch( `/api/getHistoryPerTest?testId=${testId}&limit=100`, {
+            method: 'get'
+        } );
+        const response = await fetchBuild.json();
+
+        for (let i = 0; i < response.length; i ++){
+                const testResult= response[i].tests.testResult;
+                const machine = response[i].machine;
+                if(testResult === 'PASSED') {
+                    break;
+                } else if(testResult === 'FAILED') {
+                    machines.add(machine);
+                    firstSeenFailure = response[i];
+                    machinesMap[machine] = machinesMap[machine] ? machinesMap[machine] + 1 : 1;
+                    failCount++;
+                }
+        }
+
+        let failMachineUrlBody = '';
+        machines.forEach( value => {
+            failMachineUrlBody += `Machine type: ${value}, Number: ${machinesMap[value]}\n`;
+        })
+
+        // first seen failed test details 
+        if( firstSeenFailure ) {
+            firstFailJavaVersion = firstSeenFailure.javaVersion;
+            firstFailDate = moment(firstSeenFailure.timestamp).format(DAY_FORMAT);
+            firstFailUrl = firstSeenFailure.buildUrl;
+        } 
+
+        const buildStartTime = moment(parseInt(buildTimeStamp)).format(DAY_FORMAT);
+        const title = `${testName} ${testResult} in ${buildName}`;
+        const nl = "\n";
+        const body = `**Test Info**${nl}`
+            + `Test Name: ${testName}${nl}`
+            + `Test Duration: ${renderDuration(duration)}${nl}`
+            + `Machine: ${machine}${nl}`
+            + `TRSS link for the test output: https://trss.adoptopenjdk.net/output/test${params({ id: testId })}${nl}`
+            + `${nl}${nl}`
+            + `**Build Info**${nl}`
+            + `Build Name: ${buildName}${nl}`
+            + `Jenkins Build start time: ${buildStartTime}${nl}`
+            + `Jenkins Build URL: ${buildUrl}${nl}`
+            + `TRSS link for the build: https://trss.adoptopenjdk.net/allTestsInfo${params({ buildId: buildId })}${nl}`
+            + `${nl}${nl}`
+            + `**Java Version**${nl}`
+            + `${javaVersion}${nl}`
+            + ( (firstSeenFailure == null)? `` : (
```

Comment:
Instead of using `firstSeenFailure == null `, just use `firstSeenFailure`. Please set initial value of `firstSeenFailure` to null and fix the rest of the logic.